src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SetReportingRestrictionEventHandlerTest.java

```diff
@@ -46,11 +42,14 @@ public void setupStubs() {
         dartsGateway.darNotificationReturnsSuccess();
     }
 
+    @SneakyThrows
+    @SuppressWarnings({"PMD.DoNotUseThreads"})
     @Test
     void givenSetReportingRestrictionEventReceivedAndCourtCaseAndHearingDoesNotExist_thenNotifyDarUpdate() {
 
-        var result = WireMock.listAllStubMappings();
-        log.info("mappings: {}", result.getMappings());
```

Comment:
author note:
I personally hate using `Thread.sleep`, but I can't find any better solution in this case. If anyone can suggest one I'd be happy to replace my solution

Reply:
found a better solution using awaitability